test/browser/home-page.test.jsx

```diff
@@ -4,9 +4,107 @@ var React = require('react/addons');
 var TestUtils = React.addons.TestUtils;
 
 var stubContext = require('./stub-context.jsx');
+var StubRouter = require('./stub-router');
 var HomePage = require('../../pages/home.jsx');
 
 var stubBlogFeedLoader = require('./stub-blog-feed-loader');
+var Util = require('../util.js');
+
+describe("HomePage", function() {
+ var homePage, pledgeBtn;
+
+ beforeEach(function() {
+  homePage =  stubContext.render(HomePage);
+  pledgeBtn = TestUtils.scryRenderedDOMComponentsWithClass(homePage, "icon-button")[0];
+ });
+
+ afterEach(function() {
+  stubContext.unmount(homePage);
+ });
+
+ it("does not show pledge modal by default", function() {
+  TestUtils.scryRenderedDOMComponentsWithClass(homePage, "modal-pledge").length.should.equal(0);
+ });
+
+ it("shows pledge modal when 'Pledge to Teach' button is clicked", function() {
+  TestUtils.Simulate.click(TestUtils.findRenderedDOMComponentWithTag(pledgeBtn, "a"));
+  homePage.context.showModal.callCount.should.equal(1);
```

Comment:
Oh, I believe it's because `homePage` is actually just the content that goes in the main content area of the homepage--it's not the full page, e.g. including sidebar, footer, and current modal, which is in [`components/page.jsx`](https://github.com/mozilla/teach.webmaker.org/blob/develop/components/page.jsx). In order to actually get the DOM component for the modal, you'd have to create a `Page` component that uses `HomePage` as its current route... I think you can do that by taking a look at the [`visitPage` function](https://github.com/mozilla/teach.webmaker.org/blob/develop/test/browser/page.test.jsx#L19-L27) in `page.test.jsx` if you like, but it's probably also ok to just do what you're currently doing.